jpa-crud/jpa-crud-java-sbt/jpa-crud-java-sbt-impl/src/main/java/com/example/jpacrud/application/EntityServiceImpl.java

```diff
@@ -0,0 +1,149 @@
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static java.util.stream.Collectors.toSet;
+
+public class EntityServiceImpl implements EntityService {
+
+    private static final MessageProtocol JSON = fromContentTypeHeader(of(JSON_UTF_8.toString()));
+
+    private final EntityRepository templateRepository;
+
+    @Inject
+    public EntityServiceImpl(EntityRepository templateJpaRepository) {
+        this.templateRepository = templateJpaRepository;
+    }
+
+    @Override
+    public ServiceCall<NotUsed, EntityResource> get(String id) {
+        return notUsed -> {
+            validateUUID(id);
+            return templateRepository.get(id)
+                    .thenApply(template -> toEntityResource(template.orElseThrow(notFound(id))));
+        };
+    }
+
+    @Override
+    public HeaderServiceCall<CreateEntityRequest, EntityResource> create() {
+        return (header, request) -> {
+            validate(request);
+            Entity entity = new Entity(
+                    templateRepository.nextIdentity(),
+                    request.getProperty()
+            );
+            if (request.getParts() != null) {
+                entity.setParts(request.getParts().stream().map(Part::new).collect(toSet()));
+            }
+            ResponseHeader responseHeader = new ResponseHeader(CREATED, JSON, HashTreePMap.empty());
+            return templateRepository.save(entity).thenApply(done -> Pair.create(responseHeader, toEntityResource(entity)));
+        };
+    }
+
+    @Override
+    public HeaderServiceCall<UpdateEntityRequest, NotUsed> update(String id) {
+        return (header, request) -> {
+            validateUUID(id);
+            validate(request);
+            return templateRepository.get(id)
+                    .thenCompose(t -> {
+                        Entity entity = t.orElseThrow(notFound(id));
+                        entity.setProperty(request.getProperty());
+                        if (request.getParts() != null) {
+                            entity.setParts(request.getParts().stream().map(Part::new).collect(toSet()));
+                        } else {
+                            entity.setParts(newHashSet());
+                        }
+                        return templateRepository.save(entity).thenCompose(done -> completedFuture(Pair.create(NO_CONTENT, notUsed())));
+                    });
+        };
+    }
+
+    @Override
+    public HeaderServiceCall<NotUsed, NotUsed> delete(String id) {
+        return (header, notUsed) -> {
+            validateUUID(id);
+            return templateRepository.get(id).thenCompose(template -> {
+                if (template.isPresent()) {
+                    return templateRepository.remove(template.get()).thenCompose(done -> completedFuture(Pair.create(NO_CONTENT, notUsed())));
+                } else {
+                    return completedFuture(Pair.create(NO_CONTENT, notUsed()));
+                }
+            });
+        };
+    }
```

Comment:
I'm not sure `update` and `delete` are transactional.

Reply:
I'm afraid I did not understand 😞